finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4DispatcherBase.scala

```diff
@@ -42,6 +50,33 @@ trait Netty4DispatcherBase[SendMsg <: Message, RecvMsg <: Message] {
     streams.size()
   }
 
+  @volatile private[this] var pingFramePromise: Promise[Unit] = null
+
+  protected[this] val OutstandingPingEx = Failure("Outstanding ping on HTTP/2 connection")
+  protected[this] val eventLoop = transport.context match {
+    case trans: HasExecutor => trans.executor
+    case _ => null
```

Comment:
Do we ever expect to hit this branch?  If so, consider making `eventLoop` an `Option`.  Otherwise, consider throwing a more explicit error here instead of letting this trigger an NPE somewhere else.